test/safe/branch.test.ts

```diff
@@ -171,6 +173,8 @@ describe("Branch Webhook", () => {
 			const ref = encodeURIComponent("heads/TES-123-test-ref");
 			const sha = "test-branch-ref-sha";
 
+			// mockGitHubAuthRequest();
```

Comment:
We need to mock it in any test which uses GitHub Client. 

Otherwise we rely on the test execution order. Because the reason is that this test is passing without mocking the token endpoint - the other test ("should queue and process a create webhook") gets  executed first and token stays in the Client and this test don't request it because token is already there.

This so far was the source of flakyness

Reply:
From what I tried, the sounds of the flakiness was the missing `waitUntil` for one the tests, which means the next test would run without finishing.  When I added this function in here, the tests failed because the auth request was never called.  I'm guessing this is because the Github client is never actually called in this test.